### ISSUE TITLE
Tests fail when a module in a separate project is imported

### ISSUE THREAD

**Issue by dinuish94** [ballerina-platform/ballerina-lang#19163]
**Description:**
When a module in a separate ballerina project is imported to a test class, the tests fail with the following exception.

```
[2019-09-24 11:29:55,971] SEVERE {b7a.log.crash} - org.ballerinalang.util.exceptions.BallerinaException: java.lang.NullPointerException 
org.ballerinalang.util.exceptions.BallerinaException: org.ballerinalang.util.exceptions.BallerinaException: java.lang.NullPointerException
	at org.ballerinalang.testerina.core.BTestRunner.formatErrorMessage(BTestRunner.java:819)
	at org.ballerinalang.testerina.core.BTestRunner.lambda$execute$34(BTestRunner.java:664)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at org.ballerinalang.testerina.core.BTestRunner.execute(BTestRunner.java:638)
	at org.ballerinalang.testerina.core.BTestRunner.runTest(BTestRunner.java:160)
	at org.ballerinalang.testerina.util.TesterinaUtils.executeTests(TesterinaUtils.java:132)
	at org.ballerinalang.packerina.task.RunTestsTask.execute(RunTestsTask.java:81)
	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
	at org.ballerinalang.packerina.cmd.BuildCommand.execute(BuildCommand.java:395)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.ballerinalang.tool.Main.main(Main.java:59)
Caused by: org.ballerinalang.util.exceptions.BallerinaException: java.lang.NullPointerException
	at org.ballerinalang.testerina.core.entity.TesterinaFunction.runOnSchedule(TesterinaFunction.java:146)
	at org.ballerinalang.testerina.core.entity.TesterinaFunction.invoke(TesterinaFunction.java:66)
	at org.ballerinalang.testerina.core.entity.TestSuite.start(TestSuite.java:304)
	at org.ballerinalang.testerina.core.BTestRunner.lambda$execute$34(BTestRunner.java:660)
	... 9 more
Caused by: java.lang.NullPointerException
	at sun.net.util.URLUtil.urlNoFragString(URLUtil.java:50)
	at sun.misc.URLClassPath.getLoader(URLClassPath.java:522)
	at sun.misc.URLClassPath.getNextLoader(URLClassPath.java:494)
	at sun.misc.URLClassPath.getResource(URLClassPath.java:248)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at wso2.tutorial.___init.$moduleInit(tutorial:65535)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.ballerinalang.testerina.core.entity.TesterinaFunction.lambda$runOnSchedule$0(TesterinaFunction.java:130)
	at org.ballerinalang.jvm.scheduling.SchedulerItem.execute(Scheduler.java:486)
	at org.ballerinalang.jvm.scheduling.Scheduler.run(Scheduler.java:237)
	at org.ballerinalang.jvm.scheduling.Scheduler.runSafely(Scheduler.java:207)
	at java.lang.Thread.run(Thread.java:748)
``` 

**Steps to reproduce:**

1) Build [healthcare service](https://github.com/wso2/ballerina-integrator/tree/master/docs/learn/backends/healthcare-service).

2) Move the target/caches/bir_caches/wso2/healthcare folder to <BALLERINA_HOME>_/bir_cache/wso2 to ovecome [issue](https://github.com/ballerina-platform/ballerina-lang/issues/19043)

3) Build [exposing-several-services-as-a-single-service](https://github.com/wso2/ballerina-integrator/tree/master/docs/learn/integration-tutorials/exposing-several-services-as-a-single-service) module.

**Affected Versions:**
Ballerina 1.0

**Comment by MaryamZi** [comment]
Got a similar error with 1.0.1 when trying to build a module, whose name has periods (`.`) in it.

```
[2019-10-13 17:49:26,782] SEVERE {b7a.log.crash} - java.lang.NullPointerException 
org.ballerinalang.util.exceptions.BallerinaException: java.lang.NullPointerException
	at org.ballerinalang.testerina.core.BTestRunner.lambda$processProgramFile$6(BTestRunner.java:316)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at org.ballerinalang.testerina.core.BTestRunner.processProgramFile(BTestRunner.java:308)
	at org.ballerinalang.testerina.core.BTestRunner.lambda$buildSuites$4(BTestRunner.java:285)
	at java.util.HashMap.forEach(HashMap.java:1289)
	at org.ballerinalang.testerina.core.BTestRunner.buildSuites(BTestRunner.java:273)
	at org.ballerinalang.testerina.core.BTestRunner.runTest(BTestRunner.java:156)
	at org.ballerinalang.testerina.util.TesterinaUtils.executeTests(TesterinaUtils.java:132)
	at org.ballerinalang.packerina.task.RunTestsTask.execute(RunTestsTask.java:81)
	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
	at org.ballerinalang.packerina.cmd.BuildCommand.execute(BuildCommand.java:395)
	at java.util.Optional.ifPresent(Optional.java:159)
	at org.ballerinalang.tool.Main.main(Main.java:59)
Caused by: java.lang.NullPointerException
	at sun.net.util.URLUtil.urlNoFragString(URLUtil.java:50)
	at sun.misc.URLClassPath.getLoader(URLClassPath.java:522)
	at sun.misc.URLClassPath.getNextLoader(URLClassPath.java:494)
	at sun.misc.URLClassPath.getResource(URLClassPath.java:248)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.ballerinalang.testerina.util.TestarinaClassLoader.loadClass(TestarinaClassLoader.java:67)
	at org.ballerinalang.testerina.core.BTestRunner.packageProcessed(BTestRunner.java:380)
	at org.ballerinalang.testerina.core.BTestRunner.lambda$processProgramFile$6(BTestRunner.java:311)
	... 12 more
```

**Steps to reproduce**
Create a project with a module named `foo`. 
`ballerina build foo` will not fail.

Create a module named `foo.bar` and try building it.
`ballerina build foo.bar` results in a bad, sad error.

**Comment by suganyasuven** [comment]
> **Description:**
> When a module in a separate ballerina project is imported to a test class, the tests fail with the following exception.
> 
> ```
> [2019-09-24 11:29:55,971] SEVERE {b7a.log.crash} - org.ballerinalang.util.exceptions.BallerinaException: java.lang.NullPointerException 
> org.ballerinalang.util.exceptions.BallerinaException: org.ballerinalang.util.exceptions.BallerinaException: java.lang.NullPointerException
> 	at org.ballerinalang.testerina.core.BTestRunner.formatErrorMessage(BTestRunner.java:819)
> 	at org.ballerinalang.testerina.core.BTestRunner.lambda$execute$34(BTestRunner.java:664)
> 	at java.lang.Iterable.forEach(Iterable.java:75)
> 	at org.ballerinalang.testerina.core.BTestRunner.execute(BTestRunner.java:638)
> 	at org.ballerinalang.testerina.core.BTestRunner.runTest(BTestRunner.java:160)
> 	at org.ballerinalang.testerina.util.TesterinaUtils.executeTests(TesterinaUtils.java:132)
> 	at org.ballerinalang.packerina.task.RunTestsTask.execute(RunTestsTask.java:81)
> 	at org.ballerinalang.packerina.TaskExecutor.executeTasks(TaskExecutor.java:38)
> 	at org.ballerinalang.packerina.cmd.BuildCommand.execute(BuildCommand.java:395)
> 	at java.util.Optional.ifPresent(Optional.java:159)
> 	at org.ballerinalang.tool.Main.main(Main.java:59)
> Caused by: org.ballerinalang.util.exceptions.BallerinaException: java.lang.NullPointerException
> 	at org.ballerinalang.testerina.core.entity.TesterinaFunction.runOnSchedule(TesterinaFunction.java:146)
> 	at org.ballerinalang.testerina.core.entity.TesterinaFunction.invoke(TesterinaFunction.java:66)
> 	at org.ballerinalang.testerina.core.entity.TestSuite.start(TestSuite.java:304)
> 	at org.ballerinalang.testerina.core.BTestRunner.lambda$execute$34(BTestRunner.java:660)
> 	... 9 more
> Caused by: java.lang.NullPointerException
> 	at sun.net.util.URLUtil.urlNoFragString(URLUtil.java:50)
> 	at sun.misc.URLClassPath.getLoader(URLClassPath.java:522)
> 	at sun.misc.URLClassPath.getNextLoader(URLClassPath.java:494)
> 	at sun.misc.URLClassPath.getResource(URLClassPath.java:248)
> 	at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
> 	at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
> 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
> 	at wso2.tutorial.___init.$moduleInit(tutorial:65535)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:498)
> 	at org.ballerinalang.testerina.core.entity.TesterinaFunction.lambda$runOnSchedule$0(TesterinaFunction.java:130)
> 	at org.ballerinalang.jvm.scheduling.SchedulerItem.execute(Scheduler.java:486)
> 	at org.ballerinalang.jvm.scheduling.Scheduler.run(Scheduler.java:237)
> 	at org.ballerinalang.jvm.scheduling.Scheduler.runSafely(Scheduler.java:207)
> 	at java.lang.Thread.run(Thread.java:748)
> ```
> 
> **Steps to reproduce:**
> 
> 1. Build [healthcare service](https://github.com/wso2/ballerina-integrator/tree/master/docs/learn/backends/healthcare-service).
> 2. Move the target/caches/bir_caches/wso2/healthcare folder to <BALLERINA_HOME>_/bir_cache/wso2 to ovecome [issue](https://github.com/ballerina-platform/ballerina-lang/issues/19043)
> 3. Build [exposing-several-services-as-a-single-service](https://github.com/wso2/ballerina-integrator/tree/master/docs/learn/integration-tutorials/exposing-several-services-as-a-single-service) module.
> 
> **Affected Versions:**
> Ballerina 1.0

This issue is fixed with this https://github.com/ballerina-platform/ballerina-lang/pull/20235

**Comment by parkavi11** [comment]
This issue is fixed with Ballerina version 1.0.5, hence a module from another project can be imported into a test class by directing the path in the .toml file. 
A separate issue (https://github.com/ballerina-platform/ballerina-lang/issues/20868) is created for the issue in building a module with periods.
